### PR TITLE
Rearrange partitions file structure

### PR DESCRIPTION
**Pull request by kgajdamo**
The changes was introduced to the partitioning scripts: partition_graph.py and partition_hetero_graph.py.

Proposed file structure:
- Graph partioning examples for homo and hetero graphs has been unified.
- Data by default is saved to the pytorch_geometric root directory.
- The original datasets are saved to the data/dataset/{dataset_name} folder.
- Partitioned datasets are saved to the data/partitions/{dataset_name} folder.
- The data/partitions/{dataset_name} folder contains separate subfolders due to the number of partitions, e.g.: 2-parts, 4-parts,
- in which there are separate folders for: labels, partitions (including node_map.pt, edge_map.pt and META.json etc.), test-partitions and train-partitions.
- Added the following script arguments: --use-sparse-tensor, --bf16 (support to be added).

partition-graph

### CODE REVIEW

File: examples/distributed/pyg/partition_graph.py [hunk 1]
```diff
@@ -14,32 +14,34 @@ def partition_dataset(
...
 ):
-    save_dir = osp.join(f'{root_dir}-{num_parts}parts',
-                        f'{ogbn_dataset}-partitions')
-    dataset = PygNodePropPredDataset(ogbn_dataset)
+    dataset_dir = osp.join('./data/dataset', f'{ogbn_dataset}')
+    dataset = PygNodePropPredDataset(name=ogbn_dataset, root=dataset_dir)
     data = dataset[0]
 
-    partitioner = Partitioner(data, num_parts, save_dir, recursive)
+    save_dir = osp.join(f'{root_dir}', f'{num_parts}-parts')
```
Comment (rusty1s) [inline examples/distributed/pyg/partition_graph.py:21]:
```suggestion
    save_dir = osp.join(root_dir, f'{num_parts}-parts')
```

File: examples/distributed/pyg/partition_hetero_graph.py [hunk 2]
```diff
@@ -14,30 +15,34 @@ def partition_dataset(
...
 ):
-    save_dir = osp.join(root_dir, f'{ogbn_dataset}-partitions')
-    dataset = OGB_MAG(root=ogbn_dataset, preprocess='metapath2vec')
+    dataset_dir = osp.join('./data/dataset', f'{ogbn_dataset}')
+    dataset = OGB_MAG(root=dataset_dir, preprocess='metapath2vec',
+                      transform=T.ToUndirected(merge=True))
     data = dataset[0]
 
-    partitioner = Partitioner(data, num_parts, save_dir, recursive)
+    save_dir = osp.join(f'{root_dir}', f'{num_parts}-parts')
```
Comment (rusty1s) [inline examples/distributed/pyg/partition_hetero_graph.py:23]:
```suggestion
    save_dir = osp.join(root_dir, f'{num_parts}-parts')
```

**Review by rusty1s — APPROVED** [review]

**Comment by kgajdamo** [comment]
@rusty1s please don't merge this one for now. In the meantime we came up with the idea to combine these two partitioning files. I'm in the process of changing.

File: examples/distributed/pyg/partition_graph.py [hunk 3]
```diff
@@ -3,54 +3,73 @@
 import os.path as osp
 
 import torch
-from ogb.nodeproppred import PygNodePropPredDataset
 
+from benchmark.utils import get_dataset_with_transformation
```
Comment (rusty1s) [inline examples/distributed/pyg/partition_graph.py:7]: It is a bit strange to import from `benchmark` here. Can we change this?

Comment (kgajdamo) [reply examples/distributed/pyg/partition_graph.py]: Sure, done.
